app/src/stores/permissions.ts

```diff
@@ -33,6 +40,25 @@ export const usePermissionsStore = defineStore({
 
 				return rawPermission;
 			});
+
+			function getNestedDynamicVariableFieldsInPresets(rawPermissions: Permission[]) {
+				const fields: string[] = [];
+				const rawPermissionsWithPresets = rawPermissions.filter((rawPermission: Permission) => rawPermission.presets);
+
+				for (const rawPermissions of rawPermissionsWithPresets) {
+					deepMap(rawPermissions.presets, (value) => {
+						if (typeof value !== 'string') return;
+
+						if (value.startsWith('$CURRENT_USER.')) {
+							fields.push(value.replace('$CURRENT_USER.', ''));
+						} else if (value.startsWith('$CURRENT_ROLE.')) {
+							fields.push(value.replace('$CURRENT_ROLE.', 'role.'));
+						}
+					});
```

Comment:
general logic referred from the API side here:

https://github.com/directus/directus/blob/62b4a2413ca310b1bad23c64fe37e8feac710c1e/api/src/utils/get-permissions.ts#L143-L159